### PR TITLE
Add Cody providers to pings

### PR DESCRIPTION
**Pull request by taras-yemets**
Adds Cody completions and embeddings providers data to pings.
Provider name is always included. Model name is included only for "sourcegraph" provider. 

Closes https://github.com/sourcegraph/sourcegraph/issues/58826

BigQuery pings schema update
- https://github.com/sourcegraph/analytics/pull/1028

<table>
<thead>
<tr>
<td>Site config</td>
<td>Value in pings</td>
<td>Note</td>
</tr>
</thead>
<tbody>
<tr>
<td><code>{
"cody.enabled": false
}</code></td>
<td><code>"codyProviders": {
      "Embeddings": null,
      "Completions": null
  }</code></td>
<td>If Cody is disabled, completions and embeddings are not used. Return empty values similar to <code><a href="https://github.com/sourcegraph/sourcegraph/blob/82b5f4f5d73f56e31adfe7f5bd82bf2a4206cd46/internal/conf/computed.go#L575">GetCompletionsConfig</a></code> and <code><a href="https://github.com/sourcegraph/sourcegraph/blob/82b5f4f5d73f56e31adfe7f5bd82bf2a4206cd46/internal/conf/computed.go#L816">GetEmbeddingsConfig</a></code></td>
</tr>
<tr>
<td><code>{
"cody.enabled": true,
"completions": {
    "chatModel": "claude-v1",
    "completionModel": "claude-instant-v1",
    "fastChatModel": "claude-instant-1"
    "provider": "anthropic",
    "accessToken": "REDACTED"
}
}</code></td>
<td><code>"codyProviders": {
        "Embeddings": {
            "Model": "openai/text-embedding-ada-002",
            "Provider": "sourcegraph"
        },
        "Completions": {
            "Provider": "anthropic",
            "ChatModel": "",
            "CompletionModel": "",
            "FastChatModel": ""
        }
    }</code></td>
<td>Cody is enabled, completions config is defined, embeddings - is not. Return only provider name for completions provider (don't return model names as the provider is not "sourcegraph"). Return default embeddings provider.</td>
</tr>
<tr>
<td><code>{
"cody.enabled": true,
"embeddings": {
    "provider": "sourcegraph",
    "accessToken": "REDACTED"
}
}</code></td>
<td><code>"codyProviders": {
        "Embeddings": {
            "Model": "openai/text-embedding-ada-002",
            "Provider": "sourcegraph"
        },
        "Completions": {
            "Provider": "sourcegraph",
            "ChatModel": "anthropic/claude-2.0",
            "CompletionModel": "anthropic/claude-instant-1",
            "FastChatModel": "anthropic/claude-instant-1"
        }
    }</code></td>
<td>Cody is enabled, embeddings config is defined, completions - is not. Return the provider and model name for the completions provider (note, the model name is not provided in the site config - the default one is used). Return default completions provider (with model names as the provider is "sourcegraph").</td>
</tr>
</tbody>
</table>

## Test plan
- CI passes
- Tested manually: most recent ping contains correct Cody providers data (see values in the table above)

## Preview 🤩
[Preview Link](https://docs.sourcegraph.com/@ty/cody-providers-pings)

### CODE REVIEW

File: internal/updatecheck/handler.go [hunk 1]
```diff
@@ -391,6 +392,7 @@ type pingPayload struct {
 	Timestamp                     string          `json:"timestamp"`
 	HasCodyEnabled                string          `json:"has_cody_enabled"`
 	CodyUsage                     json.RawMessage `json:"cody_usage"`
+	CodyProviders                 json.RawMessage `json:"codyProviders"`
```
Comment (chwarwick) [inline internal/updatecheck/handler.go:395]:
```suggestion
	CodyProviders                 json.RawMessage `json:"cody_providers"`
```

**Review by chwarwick — APPROVED** [review]
See note about model's otherwise I think once you get the test passing it 👍 .  Remember to wait for the data pr before merging.  Also pings used to require an approved RFC.

File: internal/usagestats/cody_providers.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package usagestats
+
+import (
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+)
+
+func GetCodyProviders() (*types.CodyProviders, error) {
+	c := conf.SiteConfig()
+	providers := types.CodyProviders{
```
[the remark this answers is not in the document]
Comment (taras-yemets) [reply internal/usagestats/cody_providers.go:11]: Makes sense!
Addressed in https://github.com/sourcegraph/sourcegraph/pull/58848/commits/b02484be53e70712cfdc4c6abb0ca5a566f3b171

**Review by chwarwick — CHANGES_REQUESTED** [review]

File: internal/usagestats/cody_providers.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/conf/conftypes"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+)
+
+func GetCodyProviders() (*types.CodyProviders, error) {
+	c := conf.SiteConfig()
+	providers := types.CodyProviders{
+		Completions: &types.CodyCompletionProvider{
+			Provider: c.Completions.Provider,
```
[the remark this answers is not in the document]
Comment (taras-yemets) [reply internal/usagestats/cody_providers.go]: Thank you, @chwarwick! Updated code and PR description.

**Review by chwarwick — APPROVED** [review]

File: client/web/src/site-admin/SiteAdminPingsPage.tsx [hunk 4]
```diff
@@ -408,6 +410,27 @@ export const SiteAdminPingsPage: React.FunctionComponent<React.PropsWithChildren
                     </li>
                     <li>Histogram of cloned repository sizes</li>
                     <li>Aggregate daily, weekly, monthly repository metadata usage statistics</li>
+                    <li>
+                        Cody providers data
+                        <ul>
+                            <li>
+                                Completions
+                                <ul>
+                                    <li>Provider</li>
```
Comment (dadlerj) [inline client/web/src/site-admin/SiteAdminPingsPage.tsx:419]: Can you clarify this a bit more? is there a discrete list of these options? If so, can you make it more clear, e.g.:

`<li>Provider (e.g. "sourcegraph", "azure-openai", "anthropic", etc.)</li>` 

or something?

In general, our ping documentation should be extremely explicit about the potential data being collected.

File: client/web/src/site-admin/SiteAdminPingsPage.tsx [hunk 5]
```diff
@@ -408,6 +410,27 @@ export const SiteAdminPingsPage: React.FunctionComponent<React.PropsWithChildren
...
+                                    <li>Chat model (included only for "sourcegraph" provider)</li>
+                                    <li>Fast chat model (included only for "sourcegraph" provider)</li>
+                                    <li>Completion model (included only for "sourcegraph" provider)</li>
+                                </ul>
+                            </li>
+                            <li>
+                                Embeddings
+                                <ul>
+                                    <li>Provider</li>
```
Comment (dadlerj) [inline client/web/src/site-admin/SiteAdminPingsPage.tsx:428]: Same

File: doc/admin/pings.md [hunk 6]
```diff
@@ -199,6 +199,13 @@ Sourcegraph aggregates usage and performance metrics for some product features i
     - Displaying ownership panel in file view.
 - Histogram of cloned repository sizes
 - Aggregate daily, weekly, monthly repository metadata usage statistics
+- Cody providers data
+  - Completions
+    - Provider name
+    - Chat, fast chat and completion model names (only for "sourcegraph" provider)
```
Comment (dadlerj) [inline doc/admin/pings.md:205]:
```suggestion
    - Chat, fast chat, and completion model names (only for "sourcegraph" provider)
```

**Review by dadlerj — APPROVED** [review]
See comments in the thread about the future of collecting this via pings: https://sourcegraph.slack.com/archives/CN4FC7XT4/p1702399336866429?thread_ts=1702377291.476959&cid=CN4FC7XT4

For now, approved!

File: client/web/src/site-admin/SiteAdminPingsPage.tsx (hunk 4 above)
Comment (taras-yemets) [reply client/web/src/site-admin/SiteAdminPingsPage.tsx]: Addressed in https://github.com/sourcegraph/sourcegraph/pull/58848/commits/0faaf3ea737ebe67e8d415809352ca06aa447313: Mentioned all currently [supported completion providers](https://github.com/sourcegraph/sourcegraph/blob/82b5f4f5d73f56e31adfe7f5bd82bf2a4206cd46/internal/conf/conftypes/consts.go#L32-L39).

**Pull request merged**